// @workspace/.ignored_api-zod/src/generated/api.ts
/**
 * Generated by orval v8.18.0 🍺
 * Do not edit manually.
 * Api
 * API specification
 * OpenAPI spec version: 0.1.0
 */
import * as zod from 'zod';


/**
 * Returns server health status
 * @summary Health check
 */
export const HealthCheckResponse = zod.object({
  "status": zod.string()
})


/**
 * @summary Get dashboard summary metrics
 */
export const GetDashboardSummaryResponse = zod.object({
  "alertsProcessed24h": zod.number(),
  "noiseReductionPct": zod.number(),
  "avgTriageSeconds": zod.number(),
  "openAlerts": zod.number(),
  "autoResolvedPct": zod.number(),
  "severityBreakdown": zod.object({
  "low": zod.number(),
  "medium": zod.number(),
  "high": zod.number(),
  "critical": zod.number()
}),
  "recentAlerts": zod.array(zod.object({
  "id": zod.number(),
  "alertId": zod.string(),
  "title": zod.string(),
  "description": zod.string(),
  "source": zod.string(),
  "severity": zod.enum(['low', 'medium', 'high', 'critical']),
  "status": zod.enum(['pending', 'true_positive', 'false_positive', 'resolved']),
  "confidence": zod.number(),
  "assetName": zod.string(),
  "mitreTactic": zod.string().nullish(),
  "createdAt": zod.coerce.date()
}))
})


/**
 * @summary List alerts in the queue
 */
export const ListAlertsQueryParams = zod.object({
  "status": zod.enum(['pending', 'true_positive', 'false_positive', 'resolved']).optional(),
  "search": zod.coerce.string().optional()
})

export const ListAlertsResponseItem = zod.object({
  "id": zod.number(),
  "alertId": zod.string(),
  "title": zod.string(),
  "description": zod.string(),
  "source": zod.string(),
  "severity": zod.enum(['low', 'medium', 'high', 'critical']),
  "status": zod.enum(['pending', 'true_positive', 'false_positive', 'resolved']),
  "confidence": zod.number(),
  "assetName": zod.string(),
  "mitreTactic": zod.string().nullish(),
  "createdAt": zod.coerce.date()
})
export const ListAlertsResponse = zod.array(ListAlertsResponseItem)


/**
 * @summary Get a single alert
 */
export const GetAlertParams = zod.object({
  "id": zod.coerce.number()
})

export const GetAlertResponse = zod.object({
  "id": zod.number(),
  "alertId": zod.string(),
  "title": zod.string(),
  "description": zod.string(),
  "source": zod.string(),
  "severity": zod.enum(['low', 'medium', 'high', 'critical']),
  "status": zod.enum(['pending', 'true_positive', 'false_positive', 'resolved']),
  "confidence": zod.number(),
  "assetName": zod.string(),
  "mitreTactic": zod.string().nullish(),
  "createdAt": zod.coerce.date()
})


/**
 * @summary Update an alert's status (triage decision)
 */
export const UpdateAlertStatusParams = zod.object({
  "id": zod.coerce.number()
})

export const UpdateAlertStatusBody = zod.object({
  "status": zod.enum(['pending', 'true_positive', 'false_positive', 'resolved'])
})

export const UpdateAlertStatusResponse = zod.object({
  "id": zod.number(),
  "alertId": zod.string(),
  "title": zod.string(),
  "description": zod.string(),
  "source": zod.string(),
  "severity": zod.enum(['low', 'medium', 'high', 'critical']),
  "status": zod.enum(['pending', 'true_positive', 'false_positive', 'resolved']),
  "confidence": zod.number(),
  "assetName": zod.string(),
  "mitreTactic": zod.string().nullish(),
  "createdAt": zod.coerce.date()
})


/**
 * @summary Simulate a batch of incoming security alerts
 */
export const simulateAlertsBodyCountMax = 20;



export const SimulateAlertsBody = zod.object({
  "count": zod.number().min(1).max(simulateAlertsBodyCountMax).optional()
})

export const SimulateAlertsResponseItem = zod.object({
  "id": zod.number(),
  "alertId": zod.string(),
  "title": zod.string(),
  "description": zod.string(),
  "source": zod.string(),
  "severity": zod.enum(['low', 'medium', 'high', 'critical']),
  "status": zod.enum(['pending', 'true_positive', 'false_positive', 'resolved']),
  "confidence": zod.number(),
  "assetName": zod.string(),
  "mitreTactic": zod.string().nullish(),
  "createdAt": zod.coerce.date()
})
export const SimulateAlertsResponse = zod.array(SimulateAlertsResponseItem)


/**
 * @summary List detected attack patterns/campaigns aggregated from alerts
 */
export const ListPatternsResponseItem = zod.object({
  "id": zod.string(),
  "name": zod.string(),
  "description": zod.string(),
  "alertCount": zod.number(),
  "severity": zod.enum(['low', 'medium', 'high', 'critical']),
  "mitreTactic": zod.string(),
  "firstSeen": zod.coerce.date(),
  "lastSeen": zod.coerce.date()
})
export const ListPatternsResponse = zod.array(ListPatternsResponseItem)


/**
 * @summary Run the 3-agent bug scanning pipeline (Analyzer, Detector, Debunker)
 */
export const RunBugScanBody = zod.object({
  "code": zod.string().optional(),
  "githubUrl": zod.string().optional(),
  "language": zod.string()
})

export const RunBugScanResponse = zod.object({
  "id": zod.number(),
  "language": zod.string(),
  "status": zod.string(),
  "analyzerNotes": zod.string(),
  "detectorFindings": zod.array(zod.object({
  "id": zod.string(),
  "title": zod.string(),
  "description": zod.string(),
  "severity": zod.enum(['low', 'medium', 'high', 'critical']),
  "line": zod.number().nullable(),
  "confirmed": zod.boolean(),
  "recommendation": zod.string()
})),
  "confirmedFindings": zod.array(zod.object({
  "id": zod.string(),
  "title": zod.string(),
  "description": zod.string(),
  "severity": zod.enum(['low', 'medium', 'high', 'critical']),
  "line": zod.number().nullable(),
  "confirmed": zod.boolean(),
  "recommendation": zod.string()
})),
  "debunkedCount": zod.number(),
  "createdAt": zod.coerce.date()
})


/**
 * @summary List recent bug scan history
 */
export const ListBugScansResponseItem = zod.object({
  "id": zod.number(),
  "language": zod.string(),
  "confirmedCount": zod.number(),
  "debunkedCount": zod.number(),
  "createdAt": zod.coerce.date()
})
export const ListBugScansResponse = zod.array(ListBugScansResponseItem)
