// @workspace/.ignored_api-client-react/src/generated/api.schemas.ts
/**
 * Generated by orval v8.18.0 🍺
 * Do not edit manually.
 * Api
 * API specification
 * OpenAPI spec version: 0.1.0
 */
export interface HealthStatus {
  status: string;
}

export type AlertStatus = typeof AlertStatus[keyof typeof AlertStatus];


export const AlertStatus = {
  pending: 'pending',
  true_positive: 'true_positive',
  false_positive: 'false_positive',
  resolved: 'resolved',
} as const;

export type AlertSeverity = typeof AlertSeverity[keyof typeof AlertSeverity];


export const AlertSeverity = {
  low: 'low',
  medium: 'medium',
  high: 'high',
  critical: 'critical',
} as const;

export interface Alert {
  id: number;
  alertId: string;
  title: string;
  description: string;
  source: string;
  severity: AlertSeverity;
  status: AlertStatus;
  confidence: number;
  assetName: string;
  /** @nullable */
  mitreTactic?: string | null;
  createdAt: string;
}

export interface AlertStatusUpdate {
  status: AlertStatus;
}

export interface SimulateAlertsInput {
  /**
     * @minimum 1
     * @maximum 20
     */
  count?: number;
}

export interface AttackPattern {
  id: string;
  name: string;
  description: string;
  alertCount: number;
  severity: AlertSeverity;
  mitreTactic: string;
  firstSeen: string;
  lastSeen: string;
}

export type DashboardSummarySeverityBreakdown = {
  low: number;
  medium: number;
  high: number;
  critical: number;
};

export interface DashboardSummary {
  alertsProcessed24h: number;
  noiseReductionPct: number;
  avgTriageSeconds: number;
  openAlerts: number;
  autoResolvedPct: number;
  severityBreakdown: DashboardSummarySeverityBreakdown;
  recentAlerts: Alert[];
}

export interface BugScanInput {
  code?: string;
  githubUrl?: string;
  language: string;
}

export interface BugScanFinding {
  id: string;
  title: string;
  description: string;
  severity: AlertSeverity;
  /** @nullable */
  line: number | null;
  confirmed: boolean;
  recommendation: string;
}

export interface BugScanResult {
  id: number;
  language: string;
  status: string;
  analyzerNotes: string;
  detectorFindings: BugScanFinding[];
  confirmedFindings: BugScanFinding[];
  debunkedCount: number;
  createdAt: string;
}

export interface BugScanSummary {
  id: number;
  language: string;
  confirmedCount: number;
  debunkedCount: number;
  createdAt: string;
}

export type ListAlertsParams = {
status?: AlertStatus;
search?: string;
};
