// @workspace/.ignored_api-client-react/src/generated/api.ts
/**
 * Generated by orval v8.18.0 🍺
 * Do not edit manually.
 * Api
 * API specification
 * OpenAPI spec version: 0.1.0
 */
import {
  useMutation,
  useQuery
} from '@tanstack/react-query';
import type {
  MutationFunction,
  QueryFunction,
  QueryKey,
  UseMutationOptions,
  UseMutationResult,
  UseQueryOptions,
  UseQueryResult
} from '@tanstack/react-query';

import type {
  Alert,
  AlertStatusUpdate,
  AttackPattern,
  BugScanInput,
  BugScanResult,
  BugScanSummary,
  DashboardSummary,
  HealthStatus,
  ListAlertsParams,
  SimulateAlertsInput
} from './api.schemas';

import { customFetch } from '../custom-fetch';
import type { ErrorType , BodyType } from '../custom-fetch';

type AwaitedInput<T> = PromiseLike<T> | T;

      type Awaited<O> = O extends AwaitedInput<infer T> ? T : never;


type SecondParameter<T extends (...args: never) => unknown> = Parameters<T>[1];



const withQueryKey = <T extends object, K>(query: T, queryKey: K): T & { queryKey: K } => {
  const result = { queryKey } as T & { queryKey: K };
  for (const key of Object.keys(query)) {
    // The explicit queryKey always wins, matching the previous
    // `{ ...query, queryKey }` spread where it was set last.
    if (key === 'queryKey') continue;
    Object.defineProperty(result, key, {
      enumerable: true,
      configurable: true,
      get: () => (query as Record<string, unknown>)[key],
    });
  }
  return result;
};

export const getHealthCheckUrl = () => {




  return `/api/healthz`
}

/**
 * Returns server health status
 * @summary Health check
 */
export const healthCheck = async ( options?: RequestInit): Promise<HealthStatus> => {

  return customFetch<HealthStatus>(getHealthCheckUrl(),
  {
    ...options,
    method: 'GET'


  }
);}





export const getHealthCheckQueryKey = () => {
    return [
    `/api/healthz`
    ] as const;
    }


export const getHealthCheckQueryOptions = <TData = Awaited<ReturnType<typeof healthCheck>>, TError = ErrorType<unknown>>( options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof healthCheck>>, TError, TData>, request?: SecondParameter<typeof customFetch>}
) => {

const {query: queryOptions, request: requestOptions} = options ?? {};

  const queryKey =  queryOptions?.queryKey ?? getHealthCheckQueryKey();



    const queryFn: QueryFunction<Awaited<ReturnType<typeof healthCheck>>> = ({ signal }) => healthCheck({ signal, ...requestOptions });





   return  { queryKey, queryFn, ...queryOptions} as UseQueryOptions<Awaited<ReturnType<typeof healthCheck>>, TError, TData> & { queryKey: QueryKey }
}

export type HealthCheckQueryResult = NonNullable<Awaited<ReturnType<typeof healthCheck>>>
export type HealthCheckQueryError = ErrorType<unknown>


/**
 * @summary Health check
 */

export function useHealthCheck<TData = Awaited<ReturnType<typeof healthCheck>>, TError = ErrorType<unknown>>(
  options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof healthCheck>>, TError, TData>, request?: SecondParameter<typeof customFetch>}

 ):  UseQueryResult<TData, TError> & { queryKey: QueryKey } {

  const queryOptions = getHealthCheckQueryOptions(options)

  const query = useQuery(queryOptions) as  UseQueryResult<TData, TError> & { queryKey: QueryKey };

  return withQueryKey(query, queryOptions.queryKey);
}







export const getGetDashboardSummaryUrl = () => {




  return `/api/dashboard/summary`
}

/**
 * @summary Get dashboard summary metrics
 */
export const getDashboardSummary = async ( options?: RequestInit): Promise<DashboardSummary> => {

  return customFetch<DashboardSummary>(getGetDashboardSummaryUrl(),
  {
    ...options,
    method: 'GET'


  }
);}





export const getGetDashboardSummaryQueryKey = () => {
    return [
    `/api/dashboard/summary`
    ] as const;
    }


export const getGetDashboardSummaryQueryOptions = <TData = Awaited<ReturnType<typeof getDashboardSummary>>, TError = ErrorType<unknown>>( options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof getDashboardSummary>>, TError, TData>, request?: SecondParameter<typeof customFetch>}
) => {

const {query: queryOptions, request: requestOptions} = options ?? {};

  const queryKey =  queryOptions?.queryKey ?? getGetDashboardSummaryQueryKey();



    const queryFn: QueryFunction<Awaited<ReturnType<typeof getDashboardSummary>>> = ({ signal }) => getDashboardSummary({ signal, ...requestOptions });





   return  { queryKey, queryFn, ...queryOptions} as UseQueryOptions<Awaited<ReturnType<typeof getDashboardSummary>>, TError, TData> & { queryKey: QueryKey }
}

export type GetDashboardSummaryQueryResult = NonNullable<Awaited<ReturnType<typeof getDashboardSummary>>>
export type GetDashboardSummaryQueryError = ErrorType<unknown>


/**
 * @summary Get dashboard summary metrics
 */

export function useGetDashboardSummary<TData = Awaited<ReturnType<typeof getDashboardSummary>>, TError = ErrorType<unknown>>(
  options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof getDashboardSummary>>, TError, TData>, request?: SecondParameter<typeof customFetch>}

 ):  UseQueryResult<TData, TError> & { queryKey: QueryKey } {

  const queryOptions = getGetDashboardSummaryQueryOptions(options)

  const query = useQuery(queryOptions) as  UseQueryResult<TData, TError> & { queryKey: QueryKey };

  return withQueryKey(query, queryOptions.queryKey);
}







export const getListAlertsUrl = (params?: ListAlertsParams,) => {
  const normalizedParams = new URLSearchParams();

  Object.entries(params || {}).forEach(([key, value]) => {

    if (value !== undefined) {
      normalizedParams.append(key, value === null ? 'null' : String(value))
    }
  });

  const stringifiedParams = normalizedParams.toString();

  return stringifiedParams.length > 0 ? `/api/alerts?${stringifiedParams}` : `/api/alerts`
}

/**
 * @summary List alerts in the queue
 */
export const listAlerts = async (params?: ListAlertsParams, options?: RequestInit): Promise<Alert[]> => {

  return customFetch<Alert[]>(getListAlertsUrl(params),
  {
    ...options,
    method: 'GET'


  }
);}





export const getListAlertsQueryKey = (params?: ListAlertsParams,) => {
    return [
    `/api/alerts`, ...(params ? [params] : [])
    ] as const;
    }


export const getListAlertsQueryOptions = <TData = Awaited<ReturnType<typeof listAlerts>>, TError = ErrorType<unknown>>(params?: ListAlertsParams, options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof listAlerts>>, TError, TData>, request?: SecondParameter<typeof customFetch>}
) => {

const {query: queryOptions, request: requestOptions} = options ?? {};

  const queryKey =  queryOptions?.queryKey ?? getListAlertsQueryKey(params);



    const queryFn: QueryFunction<Awaited<ReturnType<typeof listAlerts>>> = ({ signal }) => listAlerts(params, { signal, ...requestOptions });





   return  { queryKey, queryFn, ...queryOptions} as UseQueryOptions<Awaited<ReturnType<typeof listAlerts>>, TError, TData> & { queryKey: QueryKey }
}

export type ListAlertsQueryResult = NonNullable<Awaited<ReturnType<typeof listAlerts>>>
export type ListAlertsQueryError = ErrorType<unknown>


/**
 * @summary List alerts in the queue
 */

export function useListAlerts<TData = Awaited<ReturnType<typeof listAlerts>>, TError = ErrorType<unknown>>(
 params?: ListAlertsParams, options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof listAlerts>>, TError, TData>, request?: SecondParameter<typeof customFetch>}

 ):  UseQueryResult<TData, TError> & { queryKey: QueryKey } {

  const queryOptions = getListAlertsQueryOptions(params,options)

  const query = useQuery(queryOptions) as  UseQueryResult<TData, TError> & { queryKey: QueryKey };

  return withQueryKey(query, queryOptions.queryKey);
}







export const getGetAlertUrl = (id: number,) => {




  return `/api/alerts/${id}`
}

/**
 * @summary Get a single alert
 */
export const getAlert = async (id: number, options?: RequestInit): Promise<Alert> => {

  return customFetch<Alert>(getGetAlertUrl(id),
  {
    ...options,
    method: 'GET'


  }
);}





export const getGetAlertQueryKey = (id: number,) => {
    return [
    `/api/alerts/${id}`
    ] as const;
    }


export const getGetAlertQueryOptions = <TData = Awaited<ReturnType<typeof getAlert>>, TError = ErrorType<void>>(id: number, options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof getAlert>>, TError, TData>, request?: SecondParameter<typeof customFetch>}
) => {

const {query: queryOptions, request: requestOptions} = options ?? {};

  const queryKey =  queryOptions?.queryKey ?? getGetAlertQueryKey(id);



    const queryFn: QueryFunction<Awaited<ReturnType<typeof getAlert>>> = ({ signal }) => getAlert(id, { signal, ...requestOptions });





   return  { queryKey, queryFn, enabled: id !== null && id !== undefined, ...queryOptions} as UseQueryOptions<Awaited<ReturnType<typeof getAlert>>, TError, TData> & { queryKey: QueryKey }
}

export type GetAlertQueryResult = NonNullable<Awaited<ReturnType<typeof getAlert>>>
export type GetAlertQueryError = ErrorType<void>


/**
 * @summary Get a single alert
 */

export function useGetAlert<TData = Awaited<ReturnType<typeof getAlert>>, TError = ErrorType<void>>(
 id: number, options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof getAlert>>, TError, TData>, request?: SecondParameter<typeof customFetch>}

 ):  UseQueryResult<TData, TError> & { queryKey: QueryKey } {

  const queryOptions = getGetAlertQueryOptions(id,options)

  const query = useQuery(queryOptions) as  UseQueryResult<TData, TError> & { queryKey: QueryKey };

  return withQueryKey(query, queryOptions.queryKey);
}







export const getUpdateAlertStatusUrl = (id: number,) => {




  return `/api/alerts/${id}`
}

/**
 * @summary Update an alert's status (triage decision)
 */
export const updateAlertStatus = async (id: number,
    alertStatusUpdate: AlertStatusUpdate, options?: RequestInit): Promise<Alert> => {

  return customFetch<Alert>(getUpdateAlertStatusUrl(id),
  {
    ...options,
    method: 'PATCH',
    headers: { 'Content-Type': 'application/json', ...options?.headers },
    body: JSON.stringify(alertStatusUpdate)
  }
);}




export const getUpdateAlertStatusMutationOptions = <TError = ErrorType<void>,
    TContext = unknown>(options?: { mutation?:UseMutationOptions<Awaited<ReturnType<typeof updateAlertStatus>>, TError,{id: number;data: BodyType<AlertStatusUpdate>}, TContext>, request?: SecondParameter<typeof customFetch>}
): UseMutationOptions<Awaited<ReturnType<typeof updateAlertStatus>>, TError,{id: number;data: BodyType<AlertStatusUpdate>}, TContext> => {

const mutationKey = ['updateAlertStatus'];
const {mutation: mutationOptions, request: requestOptions} = options ?
      options.mutation && 'mutationKey' in options.mutation && options.mutation.mutationKey ?
      options
      : {...options, mutation: {...options.mutation, mutationKey}}
      : {mutation: { mutationKey, }, request: undefined};




      const mutationFn: MutationFunction<Awaited<ReturnType<typeof updateAlertStatus>>, {id: number;data: BodyType<AlertStatusUpdate>}> = (props) => {
          const {id,data} = props ?? {};

          return  updateAlertStatus(id,data,requestOptions)
        }






  return  { mutationFn, ...mutationOptions }}

    export type UpdateAlertStatusMutationResult = NonNullable<Awaited<ReturnType<typeof updateAlertStatus>>>
    export type UpdateAlertStatusMutationBody = BodyType<AlertStatusUpdate>
    export type UpdateAlertStatusMutationError = ErrorType<void>

    /**
 * @summary Update an alert's status (triage decision)
 */
export const useUpdateAlertStatus = <TError = ErrorType<void>,
    TContext = unknown>(options?: { mutation?:UseMutationOptions<Awaited<ReturnType<typeof updateAlertStatus>>, TError,{id: number;data: BodyType<AlertStatusUpdate>}, TContext>, request?: SecondParameter<typeof customFetch>}
 ): UseMutationResult<
        Awaited<ReturnType<typeof updateAlertStatus>>,
        TError,
        {id: number;data: BodyType<AlertStatusUpdate>},
        TContext
      > => {
      return useMutation(getUpdateAlertStatusMutationOptions(options));
    }

export const getSimulateAlertsUrl = () => {




  return `/api/alerts/simulate`
}

/**
 * @summary Simulate a batch of incoming security alerts
 */
export const simulateAlerts = async (simulateAlertsInput?: SimulateAlertsInput, options?: RequestInit): Promise<Alert[]> => {

  return customFetch<Alert[]>(getSimulateAlertsUrl(),
  {
    ...options,
    method: 'POST',
    headers: { 'Content-Type': 'application/json', ...options?.headers },
    body: JSON.stringify(simulateAlertsInput)
  }
);}




export const getSimulateAlertsMutationOptions = <TError = ErrorType<unknown>,
    TContext = unknown>(options?: { mutation?:UseMutationOptions<Awaited<ReturnType<typeof simulateAlerts>>, TError,{data?: BodyType<SimulateAlertsInput>}, TContext>, request?: SecondParameter<typeof customFetch>}
): UseMutationOptions<Awaited<ReturnType<typeof simulateAlerts>>, TError,{data?: BodyType<SimulateAlertsInput>}, TContext> => {

const mutationKey = ['simulateAlerts'];
const {mutation: mutationOptions, request: requestOptions} = options ?
      options.mutation && 'mutationKey' in options.mutation && options.mutation.mutationKey ?
      options
      : {...options, mutation: {...options.mutation, mutationKey}}
      : {mutation: { mutationKey, }, request: undefined};




      const mutationFn: MutationFunction<Awaited<ReturnType<typeof simulateAlerts>>, {data?: BodyType<SimulateAlertsInput>}> = (props) => {
          const {data} = props ?? {};

          return  simulateAlerts(data,requestOptions)
        }






  return  { mutationFn, ...mutationOptions }}

    export type SimulateAlertsMutationResult = NonNullable<Awaited<ReturnType<typeof simulateAlerts>>>
    export type SimulateAlertsMutationBody = BodyType<SimulateAlertsInput> | undefined
    export type SimulateAlertsMutationError = ErrorType<unknown>

    /**
 * @summary Simulate a batch of incoming security alerts
 */
export const useSimulateAlerts = <TError = ErrorType<unknown>,
    TContext = unknown>(options?: { mutation?:UseMutationOptions<Awaited<ReturnType<typeof simulateAlerts>>, TError,{data?: BodyType<SimulateAlertsInput>}, TContext>, request?: SecondParameter<typeof customFetch>}
 ): UseMutationResult<
        Awaited<ReturnType<typeof simulateAlerts>>,
        TError,
        {data?: BodyType<SimulateAlertsInput>},
        TContext
      > => {
      return useMutation(getSimulateAlertsMutationOptions(options));
    }

export const getListPatternsUrl = () => {




  return `/api/patterns`
}

/**
 * @summary List detected attack patterns/campaigns aggregated from alerts
 */
export const listPatterns = async ( options?: RequestInit): Promise<AttackPattern[]> => {

  return customFetch<AttackPattern[]>(getListPatternsUrl(),
  {
    ...options,
    method: 'GET'


  }
);}





export const getListPatternsQueryKey = () => {
    return [
    `/api/patterns`
    ] as const;
    }


export const getListPatternsQueryOptions = <TData = Awaited<ReturnType<typeof listPatterns>>, TError = ErrorType<unknown>>( options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof listPatterns>>, TError, TData>, request?: SecondParameter<typeof customFetch>}
) => {

const {query: queryOptions, request: requestOptions} = options ?? {};

  const queryKey =  queryOptions?.queryKey ?? getListPatternsQueryKey();



    const queryFn: QueryFunction<Awaited<ReturnType<typeof listPatterns>>> = ({ signal }) => listPatterns({ signal, ...requestOptions });





   return  { queryKey, queryFn, ...queryOptions} as UseQueryOptions<Awaited<ReturnType<typeof listPatterns>>, TError, TData> & { queryKey: QueryKey }
}

export type ListPatternsQueryResult = NonNullable<Awaited<ReturnType<typeof listPatterns>>>
export type ListPatternsQueryError = ErrorType<unknown>


/**
 * @summary List detected attack patterns/campaigns aggregated from alerts
 */

export function useListPatterns<TData = Awaited<ReturnType<typeof listPatterns>>, TError = ErrorType<unknown>>(
  options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof listPatterns>>, TError, TData>, request?: SecondParameter<typeof customFetch>}

 ):  UseQueryResult<TData, TError> & { queryKey: QueryKey } {

  const queryOptions = getListPatternsQueryOptions(options)

  const query = useQuery(queryOptions) as  UseQueryResult<TData, TError> & { queryKey: QueryKey };

  return withQueryKey(query, queryOptions.queryKey);
}







export const getRunBugScanUrl = () => {




  return `/api/bugscan`
}

/**
 * @summary Run the 3-agent bug scanning pipeline (Analyzer, Detector, Debunker)
 */
export const runBugScan = async (bugScanInput: BugScanInput, options?: RequestInit): Promise<BugScanResult> => {

  return customFetch<BugScanResult>(getRunBugScanUrl(),
  {
    ...options,
    method: 'POST',
    headers: { 'Content-Type': 'application/json', ...options?.headers },
    body: JSON.stringify(bugScanInput)
  }
);}




export const getRunBugScanMutationOptions = <TError = ErrorType<unknown>,
    TContext = unknown>(options?: { mutation?:UseMutationOptions<Awaited<ReturnType<typeof runBugScan>>, TError,{data: BodyType<BugScanInput>}, TContext>, request?: SecondParameter<typeof customFetch>}
): UseMutationOptions<Awaited<ReturnType<typeof runBugScan>>, TError,{data: BodyType<BugScanInput>}, TContext> => {

const mutationKey = ['runBugScan'];
const {mutation: mutationOptions, request: requestOptions} = options ?
      options.mutation && 'mutationKey' in options.mutation && options.mutation.mutationKey ?
      options
      : {...options, mutation: {...options.mutation, mutationKey}}
      : {mutation: { mutationKey, }, request: undefined};




      const mutationFn: MutationFunction<Awaited<ReturnType<typeof runBugScan>>, {data: BodyType<BugScanInput>}> = (props) => {
          const {data} = props ?? {};

          return  runBugScan(data,requestOptions)
        }






  return  { mutationFn, ...mutationOptions }}

    export type RunBugScanMutationResult = NonNullable<Awaited<ReturnType<typeof runBugScan>>>
    export type RunBugScanMutationBody = BodyType<BugScanInput>
    export type RunBugScanMutationError = ErrorType<unknown>

    /**
 * @summary Run the 3-agent bug scanning pipeline (Analyzer, Detector, Debunker)
 */
export const useRunBugScan = <TError = ErrorType<unknown>,
    TContext = unknown>(options?: { mutation?:UseMutationOptions<Awaited<ReturnType<typeof runBugScan>>, TError,{data: BodyType<BugScanInput>}, TContext>, request?: SecondParameter<typeof customFetch>}
 ): UseMutationResult<
        Awaited<ReturnType<typeof runBugScan>>,
        TError,
        {data: BodyType<BugScanInput>},
        TContext
      > => {
      return useMutation(getRunBugScanMutationOptions(options));
    }

export const getListBugScansUrl = () => {




  return `/api/bugscan/history`
}

/**
 * @summary List recent bug scan history
 */
export const listBugScans = async ( options?: RequestInit): Promise<BugScanSummary[]> => {

  return customFetch<BugScanSummary[]>(getListBugScansUrl(),
  {
    ...options,
    method: 'GET'


  }
);}





export const getListBugScansQueryKey = () => {
    return [
    `/api/bugscan/history`
    ] as const;
    }


export const getListBugScansQueryOptions = <TData = Awaited<ReturnType<typeof listBugScans>>, TError = ErrorType<unknown>>( options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof listBugScans>>, TError, TData>, request?: SecondParameter<typeof customFetch>}
) => {

const {query: queryOptions, request: requestOptions} = options ?? {};

  const queryKey =  queryOptions?.queryKey ?? getListBugScansQueryKey();



    const queryFn: QueryFunction<Awaited<ReturnType<typeof listBugScans>>> = ({ signal }) => listBugScans({ signal, ...requestOptions });





   return  { queryKey, queryFn, ...queryOptions} as UseQueryOptions<Awaited<ReturnType<typeof listBugScans>>, TError, TData> & { queryKey: QueryKey }
}

export type ListBugScansQueryResult = NonNullable<Awaited<ReturnType<typeof listBugScans>>>
export type ListBugScansQueryError = ErrorType<unknown>


/**
 * @summary List recent bug scan history
 */

export function useListBugScans<TData = Awaited<ReturnType<typeof listBugScans>>, TError = ErrorType<unknown>>(
  options?: { query?:UseQueryOptions<Awaited<ReturnType<typeof listBugScans>>, TError, TData>, request?: SecondParameter<typeof customFetch>}

 ):  UseQueryResult<TData, TError> & { queryKey: QueryKey } {

  const queryOptions = getListBugScansQueryOptions(options)

  const query = useQuery(queryOptions) as  UseQueryResult<TData, TError> & { queryKey: QueryKey };

  return withQueryKey(query, queryOptions.queryKey);
}
